### DISCUSSION TITLE
IApiResponse<T> is no longer convertible to IApiResponse

### DISCUSSION THREAD

**Discussion by Kerlione** [reactiveui/refit#1744] [Q&A] [answered]
Hello!

For some long time, I have been creating extension methods on top of IApiResponse and IApiResponse<T> to make it easier to do some validations of responses, mainly in testing scenarios.

From version 7.1.0, though, the code below no longer compiles:

```csharp
    public static void ValidateSuccess(this IApiResponse response)
    {
        response.IsSuccessStatusCode.Should().BeTrue();
        response.Error.Should().BeNull();
    }

    public static void ValidateCreated(this IApiResponse<CreationResponse> response,
        Guid expectedId)
    {
        response.ValidateSuccess();
        response.Content.Should().NotBeNull();
        response.Content!.Id.Should().Be(expectedId);
    }
```

failing with an error:

```
'IApiResponse<CreationResponse>' does not contain a definition for 'ValidateSuccess' and the best extension method overload 'IApiResponseExtensions.ValidateSuccess(IApiResponse)' requires a receiver of type 'Refit.IApiResponse'
```

Switching from IApiResponse to IApiResposeBase leads to losing the access to IsSuccessStatusCode and Error properties.
I was eager to know whether this was some expected breaking change and such usage shouldn't have been possible from the start.

**Comment by ChrisPulman** [comment, marked as answer]
I will revert the change, at the time it seemed like a reasonable update, but extensions of the original should have been taken into consideration too, my apologies. 
I will try to get a patch release out in the next couple of days.

**Reply by ChrisPulman** [reply]
@Kerlione This should be resolved now with V7.1.2

**Reply by Kerlione** [reply]
Thank you! I have checked in a local sample, that the V7.1.2 works as expected. Will try out on the actual codebase tomorrow.
